packages/build/src/github-repo.ts

```diff
@@ -60,7 +57,15 @@ export class GithubRepo {
         this.repo,
       );
 
-    return tags.find(({ commit }) => commit.sha === sha);
+    const sortedDraftTags = tags
+      .filter(t => t.name && t.name.startsWith(`v${releaseVersion}`) && t.name.match(/^v\d+\.\d+\.\d+-draft\.\d+/))
+      .map(t => ({
+        name: t.name,
+        draftVersion: semver.prerelease(t.name)?.[1] as string
+      }))
+      .filter(t => t.draftVersion !== undefined)
+      .sort((t1, t2) => parseInt(t1.draftVersion, 10) < parseInt(t2.draftVersion, 10) ? 1 : -1);
```

Comment:
I think this can just be

```suggestion
      .sort((t1, t2) => t1.draftVersion - t2.draftVersion);
```